### PR TITLE
feat(event_sources): add AWS Config Rule event data class

### LINKED ISSUE #1667: Feature request: AWS Config Rule event data class
### Use case

Lambda functions could be used by [AWS Config Custom Rules](https://docs.aws.amazon.com/config/latest/developerguide/evaluate-config_develop-rules_lambda-functions.html)
AWS Config Event (one of 3 types) will be passed as event.

### Solution/User Experience

It would be nice to have data class to work with AWS Config Rule event in the same manner, as we are working with SQS, SNS, etc.

### Alternative solutions

_No response_

### Acknowledgment

- [X] This feature request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-python/latest/#tenets)
- [X] Should this be considered in other Lambda Powertools languages? i.e. [Java](https://github.com/awslabs/aws-lambda-powertools-java/), [TypeScript](https://github.com/awslabs/aws-lambda-powertools-typescript/)

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #1667 

### Changes

In this PR we are adding the 3 types of events that can trigger a Lambda from an AWS Config rule.

These events are:

- **ConfigurationItemChangeNotification** - AWS Config publishes an event when it detects a configuration change for a resource that is within the scope of a rule.
- **ScheduledNotification** - AWS Config publishes an event when it evaluates your resources at a frequency that you specify (such as every 1 hour)
- **OversizedConfigurationItemChangeNotification** - Some resource changes generate oversized configuration items

### User experience

**BEFORE**

```python
from aws_lambda_powertools.utilities.typing import LambdaContext
from aws_lambda_powertools import Logger
import json

logger = Logger()

def lambda_handler(event, context: LambdaContext):

    message_type = json.loads(event["invokingEvent"])["messageType"]

    logger.info(f"Logging {message_type} event rule", invoke_event=event["invokingEvent"])

    return {"Success": "OK"}
```

**AFTER**

```python
from aws_lambda_powertools.utilities.data_classes import AWSConfigRuleEvent, event_source
from aws_lambda_powertools.utilities.typing import LambdaContext
from aws_lambda_powertools import Logger

logger = Logger()

@event_source(data_class=AWSConfigRuleEvent)
def lambda_handler(event: AWSConfigRuleEvent, context: LambdaContext):

    message_type = event.invoking_event.message_type

    logger.info(f"Logging {message_type} event rule", invoke_event=event.raw_invoking_event)

    return {"Success": "OK"}
```

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: docs/utilities/data_classes.md [hunk 1]
```diff
@@ -461,6 +462,26 @@ In this example, we also use the new Logger `correlation_id` and built-in `corre
     }
     ```
 
+### AWS Config Rule
+
+=== "aws_config_rule.py"
+    ```python hl_lines="1 7"
```
Comment (rubenfonseca) [inline docs/utilities/data_classes.md]:
```suggestion
    ```python hl_lines="3 11"
```

**Review by rubenfonseca — CHANGES_REQUESTED** [review]
That's a lot of great work! Just left a small change in the highlights of the docs.

File: docs/utilities/data_classes.md (hunk 1 above)
Comment (leandrodamascena) [reply docs/utilities/data_classes.md]: Done

**Comment by austindimmer** [comment]
This looks great, I was running in to this issue yesterday whilst building my first Custom Config rule and was about to build my own data structure to handle the incoming events. This looks to be exactly what is needed.

I look forward to this being merged. 

Thanks for the efforts!

**Comment by leandrodamascena** [comment]
> This looks great, I was running in to this issue yesterday whilst building my first Custom Config rule and was about to build my own data structure to handle the incoming events. This looks to be exactly what is needed.
> 
> I look forward to this being merged.
> 
> Thanks for the efforts!

Hi @austindimmer! It's amazing to hear that. We are reaching out to the AWS Config team just to confirm some information about these schemas, but we will definitely do our best to get this merged as soon as possible.

Thanks.

**Comment by leandrodamascena** [comment]
Hey, everyone! We haven't received any updates from the AWS Config team regarding event/schema validation and are moving forward with this PR.
To ensure this new `EventSource class` works as expected, we've tested extensively with many events and payloads, and in all cases, we've had success using this.

Until we get feedback from our customers that this feature is working as expected, we will not be adding Parser support.

@rubenfonseca, can you review this, please?

**Review by rubenfonseca — APPROVED** [review]
This looks amazing, thank you for the great work @leandrodamascena !

**Pull request merged**